### PR TITLE
[CI] Github actions update

### PR DESCRIPTION
**Pull request by VonOx**
Just to stay up to date ( security purpose )

### CODE REVIEW

**Comment by Pierre-Gilles** [comment]
Nice!

**Comment by VonOx** [comment]
@Pierre-Gilles This PR is ready

I've tested new docker meta action in a release build  => https://github.com/VonOx/gladys-ci/runs/3609075553

Anyway codecov report is drunk :)

File: .github/workflows/docker-release-build.yml [hunk 1]
```diff
@@ -144,15 +144,15 @@ jobs:
         uses: actions/checkout@v2
       - name: 🐳 Docker meta
         id: docker_meta
-        uses: crazy-max/ghaction-docker-meta@v1
+        uses: docker/metadata-action@v3
         with:
           images: ${{ secrets.DOCKERHUB_REPO }}
-          label-custom: |
+          labels: |
             org.opencontainers.image.title=Gladys Assitant Production Image
```
Comment (Pierre-Gilles) [inline .github/workflows/docker-release-build.yml]: Not related to this PR, but noticed a typo: "Gladys Assistant Production Image"

File: .github/workflows/docker-dev-build.yml [hunk 2]
```diff
@@ -56,10 +56,10 @@ jobs:
         uses: actions/checkout@v2
       - name: 🐳 Docker meta
         id: docker_meta
-        uses: crazy-max/ghaction-docker-meta@v1
+        uses: docker/metadata-action@v3
         with:
           images: ${{ secrets.DOCKERHUB_REPO }}
-          label-custom: |
+          labels: |
             org.opencontainers.image.title=Gladys Assitant Preprod Image
```
Comment (Pierre-Gilles) [inline .github/workflows/docker-dev-build.yml]: Not related to this PR, but there is a typo here, should be: Gladys Assistant Preprod Image

File: .github/workflows/docker-release-build.yml [hunk 3]
```diff
@@ -144,15 +144,15 @@ jobs:
...
-          tag-sha: true
-          tag-semver: |
-            {{raw}}
-            v{{major}}
+          tags: |
+            type=semver,pattern={{version}}
```
Comment (Pierre-Gilles) [inline .github/workflows/docker-release-build.yml]: We are losing the "v" here! 

See previous: 

Screenshot 2021-09-16 at 10 43 17

See your build:

Screenshot 2021-09-16 at 10 43 50

Do you confirm we are still pushing on latest as well ? thanks!

File: .github/workflows/docker-release-build.yml [hunk 4]
```diff
@@ -97,7 +97,7 @@ jobs:
         working-directory: ./server
         run: npm audit --production --audit-level=critical
       - name: 📄 Codecov report upload
-        uses: codecov/codecov-action@v1
+        uses: codecov/codecov-action@v2
```
Comment (Pierre-Gilles) [inline .github/workflows/docker-release-build.yml:100]: Not related to this PR, but we might want to add the option "fail_ci_if_error: true"

Screenshot 2021-09-16 at 10 45 36

**Review by Pierre-Gilles — CHANGES_REQUESTED** [review]
Thanks for this PR, much needed indeed! 

I noticed a few things, most are not related to the PR but it's a good moment to fix them :) 

Thanks 🙏

File: .github/workflows/docker-release-build.yml (hunk 3 above)
Comment (VonOx) [reply .github/workflows/docker-release-build.yml]: Yes latest is automatic 
image

**Comment by VonOx** [comment]
@Pierre-Gilles This PR is ready  ( review fixed )

**Comment by Pierre-Gilles** [comment]
@VonOx Looks good to me! 

Can you do a new Docker build on your repo to see if the tags generated are correct this time?

**Comment by VonOx** [comment]
@Pierre-Gilles 

https://github.com/VonOx/gladys-ci/actions/runs/1269688252
https://hub.docker.com/layers/vonox/gladys/v4.5.12/images/sha256-e3c49c24c512ccd4ef83823a8ec4dc73b5e8ea70bc6af206edff8bc104d1caad?context=repo

**Review by Pierre-Gilles — APPROVED** [review]
Looks good to me! Thanks for the PR 🙏

**Pull request merged**